src/fsharp/FSComp.txt

```diff
@@ -1428,7 +1428,7 @@ keywordDescriptionUntypedQuotation,"Delimits a untyped code quotation."
 3216,itemNotFoundDuringDynamicCodeGen,"%s '%s' not found in assembly '%s'. A possible cause may be a version incompatibility. You may need to explicitly reference the correct version of this assembly to allow all referenced components to use the correct version."
 3216,itemNotFoundInTypeDuringDynamicCodeGen,"%s '%s' not found in type '%s' from assembly '%s'. A possible cause may be a version incompatibility. You may need to explicitly reference the correct version of this assembly to allow all referenced components to use the correct version."
 descriptionWordIs,"is"
-notAFunction,"This value is not a function and cannot be applied."
+notAFunction,"This value is of type '%s', which is not a function type. A value is being passed to it as an argument as if it were a function. This is invalid."
```

Comment:
"This is invalid" - is this not implied by the fact it is an error message? You could put that on the end of any error message.

Reply:
it actually is already in ilAddressOfValueHereIsInvalid ;-)

Reply:
but yes let me remove that

Reply:
Perhaps replace with a suggested solution e.g. `This value is of type '%s', which is not a function type. A value is being passed to it as an argument as if it were a function. Consider removing the extra argument.`?

Also - quick quesiton - does this error also get provided for things like:

`foo a b` instead of `foo (a b)` ?

Reply:
![image](https://user-images.githubusercontent.com/57396/58252343-30c71380-7d66-11e9-9464-5ddf8b125fad.png)

vs.

![image](https://user-images.githubusercontent.com/57396/58252370-44727a00-7d66-11e9-9f8d-23b589eef22a.png)

I think the second one should be improved as well - but that will be much harder. Do we already have a tracking issue?

Reply:
Sorry - I meant more someone types in `foo bar 1` instead of `foo (bar 1)` .